scripts/reset-simulators.rb

```diff
@@ -17,55 +20,70 @@ def platform_for_device_type(device_type)
   end
 end
 
-def wait_for_core_simulator_service
-  # Run until we get a result since switching simulator versions often causes CoreSimulatorService to throw an exception.
-  while `xcrun simctl list devices`.empty?
-  end
-end
-
-def running_devices(devices)
-  devices.select { |device| device['state'] != 'Shutdown' }
-end
-
-def shutdown_simulator_devices(devices)
-  # Shut down any simulators that need it.
-  running_devices(devices).each do |device|
-    puts "Shutting down simulator #{device['udid']}"
-    system("xcrun simctl shutdown #{device['udid']}") or puts "    Failed to shut down simulator #{device['udid']}"
-  end
-end
-
 begin
+  # - Check if the wrong version of the CoreSimulatorService is running, stop it if so
+  if ENV['JENKINS_HOME'] || ! /^\d+\s+#{ENV['DEVELOPER_DIR']}\/.+/ =~ `/usr/bin/pgrep -lf com.apple.CoreSimulator.CoreSimulatorService 2>/dev/null` ; then    
+    # Kill all the current simulator processes as they are from a different Xcode version
+    print "Killing running Simulator processes and replacing with one from #{ENV['DEVELOPER_DIR']}..."
+    
+    if ! `/usr/bin/pgrep -qx Simulator`.empty?
+      `/usr/bin/osascript -e 'tell app "Simulator" to quit without saving'`
+      sleep 0.2 # we are sending AppleEvents, not a kill signal
```

Comment:
This offers the best chance that all of the supporting files will be in a good state.